SD_Card_Driver/Core/Src/LOS_D_sd.cpp

```diff
@@ -0,0 +1,96 @@
+#include "LOS_D_sd.hpp"
+
+int SDCard::init()
+{
+	return f_mount(&fs, "", 1);
+}
+
+int SDCard::open(const char* location)
+{
+	if (fileOpened && strcmp(location, filePath) == 0) {
+		return 0;
+	}
+
+	if (fileOpened) {
+		int res = f_close(&file);

```

Comment:
I'm not a fan of closing and reopening the same file each time we try to perform an operation on it. We should be able to just  reuse the file handle, as I have mentioned in comments below.

Reply:
The first if statement checks if the currently opened file is the file we're trying to open. If it is then the function returns without doing anything.
```
if (fileOpened && strcmp(location, filePath) == 0)
```